addons/l10n_sa_edi/models/account_edi_format.py

```diff
@@ -436,11 +436,13 @@ def _set_missing_partner_fields(missing_fields, name):
             errors.append(_set_missing_partner_fields(supplier_missing_info, _("Supplier")))
         if customer_missing_info:
             errors.append(_set_missing_partner_fields(customer_missing_info, _("Customer")))
+
         user_timezone = pytz.timezone(self.env.user.tz or 'utc')
         invoice_datetime = datetime.combine(invoice.invoice_date, datetime.min.time())
         user_invoice_datetime = user_timezone.localize(invoice_datetime)
-        utcnow = pytz.utc.localize(datetime.now())
-        if user_invoice_datetime > utcnow:
+        invoice_date_utc = user_invoice_datetime.astimezone(pytz.timezone('utc'))
+        # We should compare dates instead of datetime as it will result in inequalities.
+        if invoice_date_utc.date() > date.today():
             errors.append(_("- Please, make sure the invoice date is set to either the same as or before Today."))
```

Comment:
```suggestion
            errors.append(_("- Invoice dates cannot be set in the future."))
```
I know it's out of the scope of the commit, but this error message looks more complicated than it should :laughing: